src/charm.py

```diff
@@ -358,43 +296,25 @@ def _sync_vault_autounseal(self) -> None:
         if not self.unit.is_leader():
             logger.debug("Only leader unit can handle a vault-autounseal request")
             return
-        outstanding_requests = self.vault_autounseal_provides.get_outstanding_requests()
-        for relation in outstanding_requests:
-            self._generate_and_set_autounseal_credentials(relation)
-
-    def _set_autounseal_relation_data(
-        self, relation: Relation, key_name: str, approle_id: str, approle_secret_id: str
-    ) -> None:
-        """Set the required autounseal data in the relation databag.
-
-        Args:
-            relation: Relation for which the auto-unseal data is being set
-            key_name: The vault transit key name used for auto-unseal
-            approle_id: The AppRole ID which has permission to use this key
-            approle_secret_id: The AppRole secret ID
-        """
-        vault_address = self._get_relation_api_address(relation)
-        if not vault_address:
-            logger.warning("Vault address not available, ignoring request to set autounseal data")
-            return
-        ca_cert = (
-            self.tls.pull_tls_file_from_workload(File.CA)
-            if self.tls.ca_certificate_is_saved()
-            else None
-        )
-        if not ca_cert:
-            logger.warning("CA certificate not available, ignoring request to set autounseal data")
-            return
-
-        self.vault_autounseal_provides.set_autounseal_data(
-            relation,
-            vault_address,
-            AUTOUNSEAL_MOUNT_PATH,
-            key_name,
-            approle_id,
-            approle_secret_id,
-            ca_cert,
+        autounseal_provider_manager = VaultAutounsealProviderManager(
+            charm=self,
+            client=vault_client,
+            provides=self.vault_autounseal_provides,
+            ca_cert=self.tls.pull_tls_file_from_workload(File.CA),
+            mount_path=AUTOUNSEAL_MOUNT_PATH,
         )
+        outstanding_autounseal_requests = self.vault_autounseal_provides.get_outstanding_requests()
+        if outstanding_autounseal_requests:
+            vault_client.enable_secrets_engine(
+                SecretsBackend.TRANSIT, autounseal_provider_manager.mount_path
+            )
+        for relation in outstanding_autounseal_requests:
```

Comment:
I agree, although this is just moved code, I wrote the original and I felt bad about it. I was trying to stick with requests because that is what the other features (KV, PKI) used for their terminology.

Anyway, to fix this I made https://github.com/canonical/vault-k8s-operator/pull/550 which we will need to merge first so that the library call and the vault k8s charm is consistent.

Reply:
@ghislainbourgeois Are you fine merging this with a super promise guarentee to follow-up on https://github.com/canonical/vault-k8s-operator/pull/550 and make the same change here?